src/core/DY.Core.Label.Unknown.fst

```diff
@@ -1,3 +1,50 @@
 module DY.Core.Label.Unknown
 
-let unknown_label = Public
+#set-options "--fuel 0 --ifuel 0"
+
+let rec powf (f:pos -> pos) (steps:nat) (n:pos): pos =
+  if steps = 0 then
+    n
+  else
+    powf f (steps-1) (f n)
+
+let collatz_step (n:pos): pos =
+  if n % 2 = 0 then
+    n / 2
+  else
+    n + n + n + 1
+
+let collatz_conjecture_below (m:pos): prop =
+  forall (n:pos). n <= m ==> (exists steps. powf collatz_step steps n == 1)
+
+// `unknown_label` is corrupt only when the trace length is greater than the first counter-example of the Collatz conjecture.
+// So even if you wanted to cheat by doing `friend DY.Core.Label.Unknown`,
+// good luck reasoning with that!
+
+#push-options "--fuel 1 --ifuel 1"
+let unknown_label = {
```

Comment:
Isn't this a bit over-engineered? Defining it as public and a comment about not using `friend` would be enough, in my opinion. I think there should be at least a comment that says that this should not be used in proofs, and that's why the not provable collatz function is used. Otherwise, it probably confuses people why there is the collatz function here.

Reply:
True, I probably had too much motivation when I wrote this :)
I will simplify this.

Reply:
Done.